data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java

```diff
@@ -254,6 +254,38 @@ public String toJsonString() {
         return jsonNode.toString();
     }
 
+    /**
+     * returns a string with formatted parts replaced by their values. The input
+     * string may contain parts with format "${.../.../...}" which are replaced
+     * by their value in the event
+     * @param format string with format
+     * @throws RuntimeException if the format is incorrect or the value is not a string
+     */
+    @Override
+    public String formatString(final String format) {
+        int fromIndex = 0;
+        String result = "";
+        int pos = 0;
+        while ((pos = format.indexOf("${", fromIndex)) != -1) {
+          int endPos = format.indexOf("}", pos+1);
+          if (endPos == -1) {
+            throw new RuntimeException("index name not properly formed");
+          }
+          result += format.substring(fromIndex, pos);
+          String name = format.substring(pos+2, endPos);
+          Object val = this.get(name, String.class);
```

Comment:
This value may not always be a `String` class. I think it would be best to:

```
Object value = this.get(name, Object.class);
```

And then below on line 280:

```
result += value.toString();
```